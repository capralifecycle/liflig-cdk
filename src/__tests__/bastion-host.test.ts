import { App, Stack } from "aws-cdk-lib"
import { SecurityGroup, Vpc } from "aws-cdk-lib/aws-ec2"
import "jest-cdk-snapshot"
import { BastionHost } from ".."

test("minimal bastion-host", () => {
  const app = new App()
  const stack1 = new Stack(app, "Stack1")
  const stack2 = new Stack(app, "Stack2")

  const vpc = new Vpc(stack1, "Vpc")

  new BastionHost(stack2, "BastionHost", {
    vpc,
  })

  expect(stack2).toMatchCdkSnapshot()
})

test("bastion-host with custom security group", () => {
  const app = new App()
  const stack1 = new Stack(app, "Stack1")
  const stack2 = new Stack(app, "Stack2")

  const vpc = new Vpc(stack1, "Vpc")
  const securityGroup = new SecurityGroup(stack1, "CustomSecurityGroup", {
    vpc,
  })

  new BastionHost(stack2, "BastionHost", {
    securityGroup,
    vpc,
  })

  expect(stack2).toMatchCdkSnapshot()
})
