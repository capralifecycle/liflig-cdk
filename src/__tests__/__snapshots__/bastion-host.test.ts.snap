// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bastion-host with custom security group 1`] = `
Object {
  "Outputs": Object {
    "BastionHostBastionInstanceIdA112C7FD": Object {
      "Value": Object {
        "Ref": "BastionHostInstance82B9AD9B",
      },
    },
  },
  "Parameters": Object {
    "SsmParameterValueawsserviceamiamazonlinuxlatestamzn2amihvmx8664gp2C96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>",
    },
  },
  "Resources": Object {
    "BastionHostInstance82B9AD9B": Object {
      "DependsOn": Array [
        "BastionHostInstanceInstanceRoleDefaultPolicy662FC168",
        "BastionHostInstanceInstanceRoleA39BE04A",
      ],
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "IamInstanceProfile": Object {
          "Ref": "BastionHostInstanceInstanceProfileB1DB546D",
        },
        "ImageId": Object {
          "Ref": "SsmParameterValueawsserviceamiamazonlinuxlatestamzn2amihvmx8664gp2C96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "InstanceType": "t3.nano",
        "SecurityGroupIds": Array [
          Object {
            "Fn::ImportValue": "Stack1:ExportsOutputFnGetAttCustomSecurityGroupE5E500E5GroupId0E4C9A2D",
          },
        ],
        "SubnetId": Object {
          "Fn::ImportValue": "Stack1:ExportsOutputRefVpcPublicSubnet1Subnet5C2D37C4FFA2B456",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Bastion",
          },
        ],
        "UserData": Object {
          "Fn::Base64": Object {
            "Fn::Join": Array [
              "",
              Array [
                "#!/bin/bash
yum install -y https://amazon-ssm-",
                Object {
                  "Ref": "AWS::Region",
                },
                ".s3.amazonaws.com/latest/linux_amd64/amazon-ssm-agent.rpm socat postgresql mariadb",
              ],
            ],
          },
        },
      },
      "Type": "AWS::EC2::Instance",
    },
    "BastionHostInstanceInstanceProfileB1DB546D": Object {
      "Properties": Object {
        "Roles": Array [
          Object {
            "Ref": "BastionHostInstanceInstanceRoleA39BE04A",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "BastionHostInstanceInstanceRoleA39BE04A": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Bastion",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "BastionHostInstanceInstanceRoleDefaultPolicy662FC168": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ssm:ListInstanceAssociations",
                "ssm:UpdateInstanceInformation",
                "ssm:GetDocument",
                "ssm:PutInventory",
                "ssm:UpdateInstanceAssociationStatus",
                "ssmmessages:CreateControlChannel",
                "ssmmessages:CreateDataChannel",
                "ssmmessages:OpenControlChannel",
                "ssmmessages:OpenDataChannel",
                "ec2messages:AcknowledgeMessage",
                "ec2messages:DeleteMessage",
                "ec2messages:FailMessage",
                "ec2messages:GetEndpoint",
                "ec2messages:GetMessages",
                "ec2messages:SendReply",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BastionHostInstanceInstanceRoleDefaultPolicy662FC168",
        "Roles": Array [
          Object {
            "Ref": "BastionHostInstanceInstanceRoleA39BE04A",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;

exports[`minimal bastion-host 1`] = `
Object {
  "Outputs": Object {
    "BastionHostBastionInstanceIdA112C7FD": Object {
      "Value": Object {
        "Ref": "BastionHostInstance82B9AD9B",
      },
    },
  },
  "Parameters": Object {
    "SsmParameterValueawsserviceamiamazonlinuxlatestamzn2amihvmx8664gp2C96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>",
    },
  },
  "Resources": Object {
    "BastionHostInstance82B9AD9B": Object {
      "DependsOn": Array [
        "BastionHostInstanceInstanceRoleDefaultPolicy662FC168",
        "BastionHostInstanceInstanceRoleA39BE04A",
      ],
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "IamInstanceProfile": Object {
          "Ref": "BastionHostInstanceInstanceProfileB1DB546D",
        },
        "ImageId": Object {
          "Ref": "SsmParameterValueawsserviceamiamazonlinuxlatestamzn2amihvmx8664gp2C96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "InstanceType": "t3.nano",
        "SecurityGroupIds": Array [
          Object {
            "Fn::GetAtt": Array [
              "BastionHostSecurityGroup8CA70F42",
              "GroupId",
            ],
          },
        ],
        "SubnetId": Object {
          "Fn::ImportValue": "Stack1:ExportsOutputRefVpcPublicSubnet1Subnet5C2D37C4FFA2B456",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Bastion",
          },
        ],
        "UserData": Object {
          "Fn::Base64": Object {
            "Fn::Join": Array [
              "",
              Array [
                "#!/bin/bash
yum install -y https://amazon-ssm-",
                Object {
                  "Ref": "AWS::Region",
                },
                ".s3.amazonaws.com/latest/linux_amd64/amazon-ssm-agent.rpm socat postgresql mariadb",
              ],
            ],
          },
        },
      },
      "Type": "AWS::EC2::Instance",
    },
    "BastionHostInstanceInstanceProfileB1DB546D": Object {
      "Properties": Object {
        "Roles": Array [
          Object {
            "Ref": "BastionHostInstanceInstanceRoleA39BE04A",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "BastionHostInstanceInstanceRoleA39BE04A": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Bastion",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "BastionHostInstanceInstanceRoleDefaultPolicy662FC168": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ssm:ListInstanceAssociations",
                "ssm:UpdateInstanceInformation",
                "ssm:GetDocument",
                "ssm:PutInventory",
                "ssm:UpdateInstanceAssociationStatus",
                "ssmmessages:CreateControlChannel",
                "ssmmessages:CreateDataChannel",
                "ssmmessages:OpenControlChannel",
                "ssmmessages:OpenDataChannel",
                "ec2messages:AcknowledgeMessage",
                "ec2messages:DeleteMessage",
                "ec2messages:FailMessage",
                "ec2messages:GetEndpoint",
                "ec2messages:GetMessages",
                "ec2messages:SendReply",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BastionHostInstanceInstanceRoleDefaultPolicy662FC168",
        "Roles": Array [
          Object {
            "Ref": "BastionHostInstanceInstanceRoleA39BE04A",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BastionHostSecurityGroup8CA70F42": Object {
      "Properties": Object {
        "GroupDescription": "Stack2/BastionHost/SecurityGroup",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "Stack1:ExportsOutputRefVpc8378EB38272D6E3A",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
  },
}
`;
